tools/test.py

```diff
@@ -88,6 +93,34 @@ def merge_args(cfg, args):
         cfg.default_hooks.visualization.out_dir = args.show_dir
         cfg.default_hooks.visualization.interval = args.interval
 
+    # -------------------- badcase analyze --------------------
+    if args.badcase:
+        assert 'badcase' in cfg.default_hooks, \
+            'BadcaseAnalyzeHook is not set in the ' \
+            '`default_hooks` field of config. Please set ' \
+            '`badcase=dict(type="BadcaseAnalyzeHook")`'
+        
+        cfg.default_hooks.badcase.enable = True
+        badcase_show = cfg.default_hooks.badcase.get('show', 'False')
```

Comment:
Could we use `args.show` of `test.py` in badcase hook? I think the visualization of badcases could be a part of the visualization of prediction results

Reply:
> Could we use `args.show` of `test.py` in badcase hook? I think the visualization of badcases could be a part of the visualization of prediction results, so it seems to make sense to share the same `show`

Will it cause showing two similar images? I think the visualization of badcase covers the visualization of prediction results. If use `args.show`, should we show badcase in default or add an user option choosing which one to show?

Reply:
Yeah, I guess it will result to display twice. Is it possible to merge the badcase analysis into `PoseVisualzationHook` to enhance it? Users can choose whether to enable the badcase visualization.

Reply:
I suggest not, otherwise there will be too many args for `PoseVisualzationHook`, like `show_type(pred or badcase)`, 'store_badcase' .etc . I think we can add a `--show_type(pred or badcase)` option in argparser.

Reply:
I think the behavior should be like:
1. `show=True`, `badcase=True`, show and store the badcase, no need to show the predictions
2. `show=False`, `badcase=True`, store the badcase
3.  `True`, `False`, show the prediction results
4. `False`, 'False', nothing

Reply:
What if I just wanna analyze, store but not show badcase?

Reply:
Yeah, that refers to `2. show=False, badcase=True`, and we need to set `badcase.show = args.show`

Reply:
oh I missed that line …… This implementation logic is much greater